internal/fs/fs.go

```diff
@@ -264,6 +264,89 @@ func genTestFilename(str string) string {
 	}, str)
 }
 
+var errPathNotDir = errors.New("given path is not a directory")
+
+// ReadActualFilenames is used to determine the actual file names in given directory.
+//
+// On case sensitive file systems like ext4, it will check if those files exist using
+// `os.Stat` and return a map with key and value as filenames which exist in the folder.
+//
+// Otherwise, it reads the contents of the directory
+func ReadActualFilenames(dirPath string, names []string) (map[string]string, error) {
+	actualFilenames := make(map[string]string, len(names))
+	if len(names) <= 0 {
+		// This isn't expected to happen for current usage.
+		// Adding edge case handling, maybe useful in future
+		return actualFilenames, nil
+	}
+	// First, check that the given path is valid and it is a directory
+	dirStat, err := os.Stat(dirPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read actual filenames")
+	}
+
+	if !dirStat.IsDir() {
+		return nil, errPathNotDir
+	}
+
+	// Ideally, we would use `os.Stat` for getting the actual file names
```

Comment:
nit: i'm not sure what line length rule you're following here, and we're not really stringent about following any particular one in this project. but, at least within a particular comment block, let's be consistent 😄

Reply:
Okay. Wrapping at 80 chars.